src/ripple/basics/base_uint.h

```diff
@@ -287,7 +311,7 @@ class base_uint
         Hasher& h, base_uint const& a) noexcept
     {
         // Do not allow any endian transformations on this memory
-        h(a.pn.data (), sizeof(a.pn));
```

Comment:
I believe the intent is to catch at compile-time use of non-cryptographic hashers such as those used to store objects in unordered containers.  Instead this just allows things like the SHA-family of hashers that specify endian.  That being said, this is not the right way to say that.  If we ported rippled to a big endian machine, this constraint would fire incorrectly.

We should either require big endian, or drop the constraint entirely.

For your second question, we want the SHA hasher to treat the data as an array of bytes, and not an array of `uint32_t`.  If the hasher saw an array of `uint32_t` it might be tempted to reverse the bytes on each `uint32_t` for endian purposes.  The fact that `base_uint` stores `uint32_t` instead of `uint8_t` should be invisible to all clients, even the hasher.